Update documentation (#76)
JuliaSmoothOptimizers/PartiallySeparableNLPModels.jl: README.md

> # PartiallySeparableNLPModels: a NLPModel exploiting automatically its partially-separable structure (to define partitioned quasi-Newton models)

Reviewer: What would you think of
```suggestion
# PartiallySeparableNLPModels: An NLPModel exploiting automatically the partially-separable structure and defining partitioned quasi-Newton models
```
or I have another suggestion
```suggestion
# PartiallySeparableNLPModels: Data structures for partially-separable NLPModels and partitioned quasi-Newton models
```

Author: What about:
```suggestion
# PartiallySeparableNLPModels: Exploiting automatically the partially-separable structure to define partitioned quasi-
Newton NLPModels
```
?
After a second thought, I don't like having both NLPModels and models in the same sentence, it could be confusing.
This time, my suggestion for the title is explicitely about partitioned quasi-Newton NLPModels.
It doesn't include the partially-separable NLPModel using partial separability to compute $\nabla^2 f(x_k) v$, because it is not the main purpose of this module.
What do you think?

Reviewer: I prefer your last suggestion, thanks! Maybe we don't need `automatically` at all?

```

## How to use 

> See the [tutorial](https://JuliaSmoothOptimizers.github.io/PartiallySeparableNLPModels.jl/main/tutorial/).

Reviewer: Don't think this url works?
